pkg/repository/backend_postgres.go

```diff
@@ -773,6 +773,30 @@ func (r *PostgresBackendRepository) GetStubByExternalId(ctx context.Context, ext
 	return &stub, nil
 }
 
+func (r *PostgresBackendRepository) GetStubByExternalIdAndWorkspaceId(ctx context.Context, externalId string, workspaceId uint) (*types.StubWithRelated, error) {
```

Comment:
Can we add like an optional filter to the existing method or something? it feels like we keep adding new methods for just a slightly different query.